modules/core/src/main/scala/scalacache/memoization/MethodCallToStringConverter.scala

```diff
@@ -33,7 +33,7 @@ object MethodCallToStringConverter {
 
   private def appendClassNamePart(sb: JStringBuilder)(className: String): Unit = {
     if (className.nonEmpty) {
-      sb.append(className)
+      sb.append(className.stripSuffix("$"))
```

Comment:
Ugly... Just to retain the exact same format for keys generated for the tests. Won't even do that in general, since an `object Foo { object Bar { class Baz }}` would have `Foo$.Bar.Baz` in scala 3 rather than `Foo.Bar.Baz` - so might be worth figuring out how to rewrite the test assertions appropriately to acknowledge the slightly different className formats between scala 2 and scala 3...